### DISCUSSION TITLE
Using custom images

### DISCUSSION THREAD

**Discussion by man-of-eel** [fnuecke/oc2#104] [Q&A]
I've just built a custom Linux image with some more programs using https://github.com/fnuecke/buildroot. Can I use it, and how?

**Comment by fnuecke** [comment]
Not trivially, right now. I'm planning to allow adding these similar to the 9pfs stuff, by throwing them in a datapack. Currently the only way is to either replace the builtin one in the jar, or to create a mod/fork that registers a [BlockDeviceData](https://github.com/fnuecke/oc2/blob/1.18-forge/src/main/java/li/cil/oc2/api/bus/device/data/BlockDeviceData.java) implementation, and then spawn a `hard_drive_buildroot` item (name will change) with the `oc2:data` tag set to the name of that provider.

**Reply by man-of-eel** [reply]
Thanks. Do you know if replacing the one in the jar would break if it's bigger than 8MB?

**Reply by fnuecke** [reply]
Should work just fine!

**Reply by serpent-charmer** [reply]
Are you talking about oc2 .jar file? Can't find any trace of Image, fw_jump.bin or rootfs.ext2 in it to replace.

**Reply by fnuecke** [reply]
Ah, failed to mention that. In the sedna jar, not the oc one.
